### ISSUE TITLE
RxJava2. Incorrect behaviour at Observable#flatMap() in case of inner Observable#fromCallable() using

### ISSUE THREAD

**Issue by aleksei-tvorogov** [ReactiveX/RxJava#5687]
Hello, guys. I have faced with possible bug at RxJava2. Consider code:
```
Observabe<Boolean> someObservable = Observable.fromCallable(() -> (Boolean)null));
Observable<Boolean> complexObservable = someObservable.flatMap(someValue -> someObservable);
complexObservable.subscribe();


```
And i !didn't get! any errors about emitting null. After little researching i have found, that
_ObservableFromCallable_ implements _Callable_ interface. And inside _ObservableFlatMap_ class you can find part of code:
```
 void tryEmitScalar(Callable<? extends U> value) {
            U u;
            try {
                u = value.call();
            } catch (Throwable ex) {
                Exceptions.throwIfFatal(ex);
                errors.addThrowable(ex);
                drain();
                return;
            }

            if (u == null) {
                return;
            }
...
```

We can see, that after comparing with null, this function just returns! I think, that this is incorrect behaviour, because according docs for RxJava2 does not support null. And I think, that embeeded operators should support this requirements.
We can easy found that by default(by usual 'subscribe() call), _Observable#fromCallable()_ throws exception in case of null, but flatMap operator supress emitting null value.
Also, it is easy to understand, that replacing my 'someObservable' from code above with:
`Observabe<Boolean> someObservable = Observable.fromCallable(() -> (Boolean)null)).doOnError(Functions.emptyConsumer()))
`
 will lead to throws exception about null-returning by callable.

**Comment by akarnokd** [comment]
Please try with RxJava 2.1.5.

**Comment by akarnokd** [comment]
This bug has been already fixed in [2.1.3](https://github.com/ReactiveX/RxJava/releases/tag/v2.1.3).
